core/drivers/crypto/caam/acipher/caam_rsa.c

```diff
@@ -80,12 +80,31 @@ struct caam_rsa_keypair {
 /* CAAM Era version */
 static uint8_t caam_era;
 
+
 /*
- * Free local RSA keypair
+ * Free RSA keypair
  *
  * @key  RSA keypair
  */
-static void do_keypair_free(struct caam_rsa_keypair *key)
+static void do_keypair_free_bn(struct rsa_keypair *key)
+{
+	if (!s)
```

Comment:
This test can be removed since this function is only called if `key` isn't NULL.